app/expences.py

```diff
@@ -0,0 +1,84 @@
+import math
+import datetime
+
+from app.customers import Customer
+from app.shops import Shop
+
+
+def find_min_cost(
+        customer: Customer, shops: list[Shop], fuel_price: float
+) -> tuple | None:
+    budget = customer.money
+    print(f"{customer.name} has {budget} dollars")
+    (shop_head_to,
+     total_cost,
+     purchase_cost_min,
+     _distance
+     ) = [shops[0], 0, 0, 0]
+    for shop in shops:
+        purchase_cost = sum(
+            [shop.products[product] * customer.product_cart[product]
+             for product in customer.product_cart]
+        )
+        distance = math.sqrt(
+            (customer.location[0] - shop.location[0]) ** 2
+            + (customer.location[1] - shop.location[1]) ** 2
+        )
+        fuel_spend_cost = round(
+            customer.car["fuel_consumption"] / 100
+            * distance * 2 * fuel_price, 2)
```

Comment:
In case you do not have any arguments on the same line with the opening parenthesis, your closing parenthesis should not have any arguments as well.